aws-iot-mitigationaction/aws-iot-mitigationaction.json

```diff
@@ -0,0 +1,239 @@
+                    "type": "string",
+                    "minLength": 20,
+                    "maxLength": 2048
+                }
+            },
+            "required": [
+                "LogLevel",
+                "RoleArnForLogging"
+            ]
+        },
+        "PublishFindingToSnsParams": {
+            "type": "object",
+            "properties": {
+                "TopicArn": {
+                    "type": "string",
+                    "description": "The ARN of the topic to which you want to publish the findings.",
+                    "minLength": 20,
+                    "maxLength": 2048
+                }
+            },
+            "required": [
+                "TopicArn"
+            ]
+        },
+        "ReplaceDefaultPolicyVersionParams": {
+            "type": "object",
+            "properties": {
+                "TemplateName": {
+                    "type": "string",
+                    "enum": [
+                        "BLANK_POLICY"
+                    ]
+                }
+            },
+            "required": [
+                "TemplateName"
+            ]
+        },
+        "UpdateCACertificateParams": {
+            "type": "object",
+            "properties": {
+                "Action": {
+                    "type": "string",
+                    "enum": [
+                        "DEACTIVATE"
+                    ]
+                }
+            },
+            "required": [
+                "Action"
+            ]
+        },
+        "UpdateDeviceCertificateParams": {
+            "type": "object",
+            "properties": {
+                "Action": {
+                    "type": "string",
+                    "enum": [
+                        "DEACTIVATE"
+                    ]
+                }
+            },
+            "required": [
+                "Action"
+            ]
+        }
+    },
+    "properties": {
+        "ActionName": {
+            "description": "A unique identifier for the mitigation action.",
+            "type": "string",
+            "pattern": "[a-zA-Z0-9:_-]+",
+            "minLength": 1,
+            "maxLength": 128
+        },
+        "RoleArn": {
+            "type": "string"
+        },
+        "Tags": {
+            "type": "array",
+            "maxItems": 50,
+            "uniqueItems": true,
+            "insertionOrder": false,
+            "description": "An array of key-value pairs to apply to this resource.",
+            "items": {
+                "$ref": "#/definitions/Tag"
+            }
+        },
+        "ActionParams": {
+            "$ref": "#/definitions/ActionParams"
+        },
+        "MitigationActionArn": {
+            "type": "string"
+        },
+        "MitigationActionId": {
+            "type": "string"
+        },
+        "CreationDate": {
+            "type": "string"
+        }
+    },
+    "additionalProperties": false,
+    "required": [
+        "ActionName",
```

Comment:
Shouldn't the ActionName be optional for this schema? CreateHandler generates an ActionName if it's not provided.

Reply:
Good catch! Changed.